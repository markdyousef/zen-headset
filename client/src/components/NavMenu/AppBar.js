import React from "react";
import AppBar from "@material-ui/core/AppBar";
import ToolBar from "@material-ui/core/Toolbar";
import IconButton from "@material-ui/core/IconButton";
import MenuIcon from "@material-ui/icons/Menu";
import Typography from "@material-ui/core/Typography";
import AccountCircle from "@material-ui/icons/AccountCircle";
import Button from "@material-ui/core/Button";
import styled from "styled-components";
import {NavLink} from "react-router-dom";

const AppBarContainer = styled.section`
  padding-bottom: 60px;
`;

const Link = styled(NavLink)`
  text-decoration: none;
  color: inherit;
`

export default ({ changeDrawer, changeDialog }) => {
  return (
    <AppBarContainer>
      <AppBar>
        <ToolBar>
          <IconButton>
            <MenuIcon onClick={() => changeDrawer()} />
          </IconButton>
          <Typography variant="title" style={{ flex: 1 }}>
            {/* Discover */}
          </Typography>
          <Button variant="contained" color="primary" onClick={() => changeDialog()}>Connect</Button>
          <IconButton>
            <Link to="/account"><AccountCircle /></Link>
          </IconButton>
        </ToolBar>
      </AppBar>
    </AppBarContainer>
  );
};
