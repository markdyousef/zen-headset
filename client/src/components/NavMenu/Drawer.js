import React from 'react';
import Drawer from "@material-ui/core/Drawer";
import List from "@material-ui/core/List";
import ListItem from "@material-ui/core/ListItem";
import ListItemText from "@material-ui/core/ListItemText";
import WorldIcon from "@material-ui/icons/Language";
import Divider from "@material-ui/core/Divider";
import styled from "styled-components";

const ListContainer = styled.nav`
  width: 200px;
`;

const NavListItem = ({ onChange }) => (
  <ListContainer>
    <List>
      <ListItem button>
        <WorldIcon />
        <ListItemText primary="Discover" onClick={() => onChange(false)} />
      </ListItem>
    </List>
  </ListContainer>
);

export default ({open, onChange}) => (
  <Drawer open={open} onClose={() => onChange(false)}>
    <NavListItem onChange={onChange} />
  </Drawer>
);
