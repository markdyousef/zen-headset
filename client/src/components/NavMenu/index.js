import React, { Component } from "react";
import AppBar from "./AppBar";
import Drawer from "./Drawer";
import HeadsetDialog from "../HeadsetDialog";

class NavMenu extends Component {
  state = {
    showDrawer: false,
    showDialog: false
  };
  handleDrawerChange = (value = true) => {
    this.setState({ showDrawer: value });
  };
  handleDialogChange = (value = true) => {
    console.log("Dialog")
    console.log(value)
    this.setState({ showDialog: value });
  };
  render() {
    return (
      <div>
        <AppBar 
          changeDrawer={this.handleDrawerChange}
          changeDialog={this.handleDialogChange}
        />
        <Drawer
          open={this.state.showDrawer}
          onChange={this.handleDrawerChange}
        />
        {this.state.showDialog&&<HeadsetDialog
          open={this.state.showDialog}
          onClose={() => this.handleDialogChange(false)}
        />}
      </div>
    );
  }
}

export default NavMenu;
