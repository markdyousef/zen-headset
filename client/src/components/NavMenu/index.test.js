import React from "react";
import { shallow } from "enzyme";
import NavMenu from "./index";

describe("NavMenu", () => {
  test("should match snapshot", () => {
    const wrapper = shallow(<NavMenu />);
    expect(wrapper).toMatchSnapshot();
  });
});
