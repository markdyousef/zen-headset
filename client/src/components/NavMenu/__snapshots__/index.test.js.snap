// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavMenu should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NavMenu />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <WithStyles(AppBar)>
          <WithStyles(Toolbar)>
                    <WithStyles(IconButton)>
                              <pure(Menu)
                                        onClick={[Function]}
                              />
                    </WithStyles(IconButton)>
                    <WithStyles(Typography)
                              variant="title"
                    >
                              Title
                    </WithStyles(Typography)>
          </WithStyles(Toolbar)>
</WithStyles(AppBar)>,
        <WithStyles(Drawer)
          onClose={[Function]}
          open={false}
>
          <NavListItem
                    onChange={[Function]}
          />
</WithStyles(Drawer)>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <WithStyles(Toolbar)>
            <WithStyles(IconButton)>
                        <pure(Menu)
                                    onClick={[Function]}
                        />
            </WithStyles(IconButton)>
            <WithStyles(Typography)
                        variant="title"
            >
                        Title
            </WithStyles(Typography)>
</WithStyles(Toolbar)>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <WithStyles(IconButton)>
                <pure(Menu)
                                onClick={[Function]}
                />
</WithStyles(IconButton)>,
              <WithStyles(Typography)
                variant="title"
>
                Title
</WithStyles(Typography)>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <pure(Menu)
                  onClick={[Function]}
/>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Title",
                "variant": "title",
              },
              "ref": null,
              "rendered": "Title",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <NavListItem
            onChange={[Function]}
/>,
          "onClose": [Function],
          "open": false,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "onChange": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <WithStyles(AppBar)>
            <WithStyles(Toolbar)>
                        <WithStyles(IconButton)>
                                    <pure(Menu)
                                                onClick={[Function]}
                                    />
                        </WithStyles(IconButton)>
                        <WithStyles(Typography)
                                    variant="title"
                        >
                                    Title
                        </WithStyles(Typography)>
            </WithStyles(Toolbar)>
</WithStyles(AppBar)>,
          <WithStyles(Drawer)
            onClose={[Function]}
            open={false}
>
            <NavListItem
                        onChange={[Function]}
            />
</WithStyles(Drawer)>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <WithStyles(Toolbar)>
              <WithStyles(IconButton)>
                            <pure(Menu)
                                          onClick={[Function]}
                            />
              </WithStyles(IconButton)>
              <WithStyles(Typography)
                            variant="title"
              >
                            Title
              </WithStyles(Typography)>
</WithStyles(Toolbar)>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <WithStyles(IconButton)>
                  <pure(Menu)
                                    onClick={[Function]}
                  />
</WithStyles(IconButton)>,
                <WithStyles(Typography)
                  variant="title"
>
                  Title
</WithStyles(Typography)>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <pure(Menu)
                    onClick={[Function]}
/>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Title",
                  "variant": "title",
                },
                "ref": null,
                "rendered": "Title",
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <NavListItem
              onChange={[Function]}
/>,
            "onClose": [Function],
            "open": false,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "onChange": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
